### PR TITLE
Removes server header from cowboy applications

### PR DESCRIPTION
**Pull request by dunson062786**
https://github.com/smartcolumbusos/scosopedia/issues/148

This code removes the server headers from our cowboy applications.

### CODE REVIEW

**Comment by dunson062786** [comment]
sure thing @bbrewer-papa-x. Should I put the common file in https://github.com/smartcitiesdata/smart_city/tree/master/lib/smart_city?

File: apps/web/lib/web.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
+defmodule Web do
```
Comment (LtChae) [inline apps/web/lib/web.ex:1]: Should probably delete this.

File: apps/web/lib/web/application.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+defmodule Web.Application do
+  # See https://hexdocs.pm/elixir/Application.html
+  # for more information on OTP Applications
+  @moduledoc false
+
+  use Application
+
+  def start(_type, _args) do
```
Comment (LtChae) [inline apps/web/lib/web/application.ex:8]: Does this need to be a supervised application or is it just a library?

File: apps/web/mix.exs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    [
+      extra_applications: [:logger],
+      mod: {Web.Application, []}
+    ]
+  end
+
+  # Run "mix help deps" to learn about dependencies.
+  defp deps do
+    [
+      # {:dep_from_hexpm, "~> 0.3.0"},
```
Comment (LtChae) [inline apps/web/mix.exs:29]: Clean up boilerplate where possible

File: apps/web/test/web_test.exs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+defmodule WebTest do
```
Comment (LtChae) [inline apps/web/test/web_test.exs:1]: Should probably have actual tests instead of a placeholder

File: apps/web/lib/web/application.ex (hunk 2 above)
Comment (dunson062786) [reply apps/web/lib/web/application.ex:8]: It's is just a library

Comment (LtChae) [reply apps/web/lib/web/application.ex:8]: It doesn't need this application file then. See the Providers library in apps.

File: apps/web/test/web_test.exs (hunk 4 above)
Comment (dunson062786) [reply apps/web/test/web_test.exs:1]: I'm not sure how I would test this. There is only one function in strip_server_header that has a non-default implementation and that is the info method. It does a Map.drop on the header field. I could create an implementation class that does a Map.drop() and then test that and have info call that function.

File: apps/web/mix.exs (hunk 3 above)
Comment (dunson062786) [reply apps/web/mix.exs]: How does it look now?

File: apps/web/lib/web/application.ex (hunk 2 above)
Comment (dunson062786) [reply apps/web/lib/web/application.ex]: I've deleted the application.ex file

File: apps/web/lib/web.ex (hunk 1 above)
Comment (dunson062786) [reply apps/web/lib/web.ex]: done

File: apps/web/mix.exs (hunk 3 above)
Comment (dunson062786) [reply apps/web/mix.exs]: I've deleted the boilerplate

File: apps/andi/test/integration/andi_web/controllers/dataset_controller_test.exs [hunk 5]
```diff
@@ -356,6 +356,14 @@ defmodule Andi.CreateDatasetTest do
     end
   end
 
+  describe "dataset get" do
+    test "andi doesn't return server in response headers" do
+      {:ok, %Tesla.Env{headers: headers}} = get("/api/v1/datasets", headers: [{"content-type", "application/json"}])
+      IO.puts("headers is #{inspect(headers)}")
```
Comment (dunson062786) [inline apps/andi/test/integration/andi_web/controllers/dataset_controller_test.exs:362]: I'll remove this

File: apps/andi/test/integration/andi_web/controllers/dataset_controller_test.exs [hunk 6]
```diff
@@ -356,6 +356,14 @@ defmodule Andi.CreateDatasetTest do
     end
   end
 
+  describe "dataset get" do
+    test "andi doesn't return server in response headers" do
+      {:ok, %Tesla.Env{headers: headers}} = get("/api/v1/datasets", headers: [{"content-type", "application/json"}])
```
Comment (bbrewer-papa-x) [inline apps/andi/test/integration/andi_web/controllers/dataset_controller_test.exs:361]: The test itself is fine, but I would suggest either pattern matching out the tuple you care about (assuming you know exactly what headers are coming back) or turning it into a map and checking if it has the `server` key:
```suggestion
      {:ok, %Tesla.Env{headers: headers}} = get("/api/v1/datasets", headers: [{"content-type", "application/json"}])
      refute headers |> Map.new() |> Map.has_key?("server")
```

File: apps/web/test/web_test.exs (hunk 4 above)
Comment (dunson062786) [reply apps/web/test/web_test.exs]: I've written integration test for andi, discovery_api, and discovery_streams to test this functionality.

**Pull request merged**